ledger-core/testutils/recordchecker/checker.go

```diff
@@ -0,0 +1,314 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package recordchecker
+
+import (
+	"reflect"
+	"time"
+
+	"github.com/gojuno/minimock/v3"
+
+	"github.com/insolar/assured-ledger/ledger-core/pulse"
+	"github.com/insolar/assured-ledger/ledger-core/reference"
+	"github.com/insolar/assured-ledger/ledger-core/rms"
+	"github.com/insolar/assured-ledger/ledger-core/rms/rmsbox"
+	"github.com/insolar/assured-ledger/ledger-core/vanilla/throw"
+)
+
+type Constraint func(actualMsg rms.LRegisterRequest) error
+type ResponseProducer func(message rms.LRegisterRequest)
+
+type Checker interface {
+	// Add new chainItem of expected messages
+	NewChainFromReference(objectRef rms.Reference) Chain
```

Comment:
Checker does New? naming is very confusing

Reply:
complex renaming of checker and other interfaces/methods